io_scene_godot/converters/multimesh.py

```diff
@@ -0,0 +1,185 @@
+        self.particle_system = particle_system
+
+        self.mesh_resource = None
+
+    def export_multimesh(self, escn_file, export_settings, particle_name):
+        """Saves a mesh into the escn file"""
+        converter = MultiMeshConverter(self.particle_system)
+        key = MultiMeshResourceKey(
+            'MultiMesh', self.object, export_settings, particle_name)
+        # Check if mesh resource exists so we don't bother to export it twice,
+        multimesh_id = escn_file.get_internal_resource(key)
+        if multimesh_id is not None:
+            return multimesh_id
+
+        multimesh = converter.to_multimesh()
+        if multimesh is not None:
+            self.mesh_resource = MultiMeshResource(particle_name)
+            self.mesh_resource['instance_count'] = '{}'.format(
+                len(self.particle_system.particles))
+            self.mesh_resource['mesh'] = 'SubResource({})'.format(
+                self.instance_mesh_id)
+            self.mesh_resource['transform_array'] = (
+                'PoolVector3Array({})'.format(
+                    converter.to_multimesh())
+                )
+
+            multimesh_id = escn_file.add_internal_resource(
+                self.mesh_resource, key)
+            assert multimesh_id is not None
+
+        return multimesh_id
+
+
+class MultiMeshResourceKey:
+    """Produces a key based on an mesh object's data, every different
+    Mesh Resource would have a unique key"""
+
+    def __init__(self, rsc_type, obj, export_settings, particle_name):
+        mesh_data = obj.data
+
+        # Resource type included because same blender mesh may be used as
+        # MeshResource or CollisionShape, but they are different resources
+        gd_rsc_type = rsc_type
+
+        # Precalculate the hash now for better efficiency later
+        self._data = tuple([obj.name, particle_name, gd_rsc_type])
+        self._hash = hash(self._data)
+
+    def __hash__(self):
+        return self._hash
+
+    def __eq__(self, other):
+        # pylint: disable=protected-access
+        return (self.__class__ == other.__class__ and
+                self._data == other._data)
+
+
+class MultiMeshResource(InternalResource):
+    """Godot MultiMesh resource"""
+
+    def __init__(self, name):
+        super().__init__('MultiMesh', name)
+        self['transform_format'] = 1
+
+        # Change above value in MultiMeshResourceExporter
+        self['instance_count'] = 0
+        self['mesh'] = None
+        self['transform_array'] = None
+
+
+class MultiMeshConverter:
+    """Blender Particles' mat4x4 to
+    Godot MultiMesh resource PoolVector3Array"""
+
+    def __init__(self, particle_system):
+        self.particle_system = particle_system
+
+    def to_multimesh(self):
+        transform_array = []
+        float32array = ''
+        for p in self.particle_system.particles:
+            quat_x = mathutils.Quaternion((1.0, 0.0, 0.0), math.radians(90.0))

```

Comment:
Sorry, I think I need some context here 😅
Are these quaternion operations handles the up-axis difference between Godot and Blender?

Reply:
no,its' not difference between Godot and Blender,but is difference between Blender's particle and Blender's object,i have tested many times,i thought this may because particles in Blender are very low level and computed directly by GPU,dont know too much about that, just guess....

Reply:
Okay. We can use the current implementation. We will figure it out when people start to use this feature.